### PR TITLE
Add Events System to ECS

### PR DESCRIPTION
**Pull request by lucasrumney94**
Adding Events using Box Collision as the first system to use it

### CODE REVIEW

File: OpenGL Template/BoxColliderSystem.hpp [hunk 1]
```diff
@@ -15,14 +16,24 @@
 #include <glm\gtc\matrix_transform.hpp>
 #include <cmath>
 
+typedef void (*CollisionEventListener) (Entity*, Entity*);
```
Comment (lucasrumney94) [inline OpenGL Template/BoxColliderSystem.hpp]: box?

File: OpenGL Template/BoxColliderSystem.cpp [hunk 2]
```diff
@@ -1,10 +1,15 @@
 #include "BoxColliderSystem.hpp"
-#include "BoxCollider.hpp"
-#include "Transform.hpp"
 
 extern Coordinator gCoordinator;
 
+void BoxCollisionEvent::RaiseBoxCollisionEvent(Entity* owner, Entity* other) {
+	for (CollisionEventListener listener : Subscribers) {
```
Comment (lucasrumney94) [inline OpenGL Template/BoxColliderSystem.cpp]: box?

File: OpenGL Template/RenderSystem.hpp [hunk 3]
```diff
@@ -16,8 +16,6 @@
 #include <glm\gtc\matrix_transform.hpp>
 #include <cmath>
 
-class Event;
```
Comment (lucasrumney94) [inline OpenGL Template/RenderSystem.hpp:19]: 😅

File: OpenGL Template/main.cpp [hunk 4]
```diff
@@ -10,6 +10,8 @@
 #include "RenderSystem.hpp"
 #include "BoxColliderSystem.hpp"
 
+#include "Event.hpp"
+
```
Comment (lucasrumney94) [inline OpenGL Template/main.cpp:14]: should we take out the empty lines

**Review by lucasrumney94 — COMMENTED** [review]
small updates needed, but amazing progress guys!

**Review by htw6174 — APPROVED** [review]

**Pull request merged**